nni/compression/experiment/config/compression.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Type, Union
+
+from torch.nn import Module
+
+from nni.experiment.config.base import ConfigBase
+from .pruner import PrunerConfig
+from .quantizer import QuantizerConfig
+from .common import ComparableType
+
+__all__ = ['CompressionConfig']
+
+
+@dataclass
+class CompressionConfig(ConfigBase):
+    # constraints
+    params: Union[str, int, float, List[Union[str, int, float]], None] = None
+    flops: Union[str, int, float, List[Union[str, int, float]], None] = None
+    # latency: float, List[float], None
+    metric: Union[ComparableType, List[ComparableType], None] = None
+
+    # compress scope description
+    module_types: Optional[List[Union[Type[Module], Dict[Type[Module], Dict], str, Dict[str, Dict]]]] = None
+    module_names: Union[List[str], Dict[str, Dict], None] = None
+    exclude_module_names: Union[List[str], Dict[str, Dict], None] = None
```

Comment:
What does `Dict[str, Dict]` means here?

Reply:
some specific configures like `{'module_name_a': {'sparsity_range': [0, 0.5]}}`
now, only support `['module_name_a']`
I will del these unused type hints for now.

Reply:
Then I suggest to add a type alias.

Reply:
yes